core/services/cable_guy/api/manager.py

```diff
@@ -250,23 +265,14 @@ def set_static_ip(self, interface_name: str, ip: str) -> None:
         # Set new ip address
         self.set_ip(interface_name, ip)
 
-    def get_interfaces(self) -> List[Dict[str, Any]]:
+    def get_interfaces(self) -> List[EthernetInterface]:
         """Get interfaces information
 
         Returns:
-            dict: Interface information that uses the following struct:
+            List of EthernetInterface objects available:
             [
-                {
-                    'name': 'interface_name',
-                    'configuration': {
-                        'ip': 'ip_address',
-                        'mode': 'mode' // [unmanaged, client, server],
-                    },
-                    'info': {
-                        'connected': True,
-                        'number_of_disconnections': 4,
-                    }
-                },
+                interface1,
```

Comment:
The list structure is not important to document since is already described in the documentation as a list of a structure type.

Reply:
Agree.

Do you know if typehints get to the docstrings somehow?

Reply:
No deia.